src/overlays/actors/ovl_En_Ex_Ruppy/z_en_ex_ruppy.c

```diff
@@ -21,29 +34,345 @@ const ActorInit En_Ex_Ruppy_InitVars = {
+            }
+        } else {
+            //That idiot! error
+            osSyncPrintf(VT_FGCOL(GREEN) "☆☆☆☆☆ そ、そんなばかな！エラー！！！！！ ☆☆☆☆☆ \n" VT_RST);
+        }
+        //Stupid!
+        osSyncPrintf(VT_FGCOL(GREEN) "☆☆☆☆☆ バカめ！ ☆☆☆☆☆ \n" VT_RST);
+        tempExplosionScale = 100;
+        tempExplosionScaleStep = 30;
+        if (this->unk_152 == 2) {
+            tempExplosionScale = 20;
+            tempExplosionScaleStep = 6;
+        }
+        EffectSsBomb2_SpawnLayered(globalCtx, &this->actor.posRot.pos, &tempZeroVector, &tempPoint1Vec, tempExplosionScale, tempExplosionScaleStep);
+        func_8002F71C(globalCtx, &this->actor, 2.0f, this->actor.yawTowardsLink, 0.0f);
+        Audio_PlayActorSound2(&this->actor, NA_SE_IT_BOMB_EXPLOSION);
+        Actor_Kill(&this->actor);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ex_Ruppy/func_80A0AF24.s")
+void EnExRuppy_Collect(EnExRuppy *this, GlobalContext *globalCtx) { 
+    const f32 localConst = 30.0f;
+    if (this->actor.xyzDistFromLinkSq < SQ(localConst)) {
+        func_80078884(NA_SE_SY_GET_RUPY);
+        Item_DropCollectible(globalCtx, &this->actor.posRot.pos, ((s16)D_80A0B320[this->unk_150 ] | 0x8000));
+        Actor_Kill(&this->actor);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ex_Ruppy/func_80A0B070.s")
+void func_80A0B0F4(EnExRuppy *this, GlobalContext *globalCtx) {
+    if (this->unk_15C != 0) {
+        Math_SmoothScaleMaxF(&this->actor.shape.unk_08, 700.0f, 0.5f, 200.0f);
+    } else {
+    Math_SmoothScaleMaxF(&this->actor.shape.unk_08, -700.0f, 0.5f, 200.0f);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ex_Ruppy/func_80A0B0F4.s")
+void EnExRuppy_Update(Actor *thisx, GlobalContext *globalCtx) {
+    EnExRuppy* this = THIS;
+    thisx->shape.rot.y += 1960;
+    this->actionFunc(this,globalCtx);
+    if (this->timer != 0) {
+        this->timer--;
+    }
+    Actor_MoveForward(&this->actor);
+    func_8002E4B4(globalCtx, &this->actor, 20.0f, 20.0f, 50.0f, 0x1C);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ex_Ruppy/EnExRuppy_Update.s")
+void EnExRuppy_Draw(Actor *thisx, GlobalContext *globalCtx) {
+    EnExRuppy* this = THIS;
+    GraphicsContext* gfxCtx;
+    s32 pad;
+    Gfx* dispRefs[4];
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ex_Ruppy/EnExRuppy_Draw.s")
+    if (this->isFalling == 0) {
+        gfxCtx = globalCtx->state.gfxCtx;
+        Graph_OpenDisps(dispRefs, globalCtx->state.gfxCtx, "../z_en_ex_ruppy.c", 774);
+        func_80093D18(globalCtx->state.gfxCtx);
+        func_8002EBCC(thisx, globalCtx, 0);
+        gSPMatrix(gfxCtx->polyOpa.p++, Matrix_NewMtx(globalCtx->state.gfxCtx,"../z_en_ex_ruppy.c", 780), G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+        gSPSegment(gfxCtx->polyOpa.p++, 0x08, SEGMENTED_TO_VIRTUAL(D_80A0B3B8[this->unk_150]));
+        gSPDisplayList(gfxCtx->polyOpa.p++, &D_04042440);
```

Comment:
no & on the array

Reply:
also, is this even externed in here? how is this compiling?
make sure to declare it with gfx[] type

Reply:
Apparently its defined somewhere else and it doesn't need to be defined here too.

Reply:
Also not using the & caused issues.
![image](https://user-images.githubusercontent.com/35883445/90592779-67334e80-e1b4-11ea-85c4-b41423cb13b1.png)

Reply:
this is cause its in variables.h typed as a u32. fix it to gfx[]